docs/syntax-named-function-type-returns.md

```diff
@@ -0,0 +1,197 @@
+local blue, red = doStuff()
+
+-- lint: `doStuff` sends identically named & typed values to `receiveStuff`, but in the wrong order
+receiveStuff(doStuff())
+```
+
+
+
+## Drawbacks
+
+There is a philosophical disagreement over the purpose of the Luau static type system:
+- This proposal was written on the grounds that Luau should have internal syntactic consistency and cross-pollination. Runtime, type checking, and documentation features should enmesh and enrich each other as a unified whole, to ensure everything is kept in sync and is easily comprehensible.
+- The primary argument against this proposal is the belief that Luau's type checker should be a "pure proof system", cleanly separated from documentation concerns at every point, so that the type system can exist in a pure realm unconcerned with semantic information.
+
+This feels like a core tension about the direction of type checking features as a whole. We should check in with the community on this.
+
+There is a particular concern that introducing comprehension aids into the type language would imply meaning where there is none. Would users think that return names have functional meaning?
+
+There are other places in Luau where we have previously introduced comprehension aids in the past:
+
+```Lua
+-- number literals could have been kept pure, but we chose to add meaningless delimiters for easier comprehension
+local ONE_BILLION = 1000000000
+local ONE_BILLION = 1_000_000_000
+
+-- generics could have been referenced abstractly, but we chose to give them human-readable names
+type Foo<2> = (<1>, <2>) -> <1> | <2>
+type Foo<OK, Fallback> = (OK, Fallback) -> OK | Fallback
+
+-- parameters in function types didn't have to have named parameters, but we decided to add them
+type Func = (foo: number, bar: number) -> ()
+```
+
+In particular, it is established convention that Luau does not rearrange the contents of lists bounded by parentheses, even when list items are named:
+
+```Lua
+local function poem(red: number, blue: number)
+    print("roses are", red, "violets are", blue)
+end
+
+local blue, red = "blue", "red"
+
+poem(blue, red) --> roses are blue violets are red
+```
+
+So, this proposal posits that there is already established precedent for such features, and that users understand how comprehension aids function in Luau today.
+
+A common concern is whether these comprehension aids would mislead people into believing that names are significant when considering type compatibility.
+
+However, we already have features that clearly indicate we don't do this.
+
+```Lua
+-- names of generics are ignored
+type Foo = <A, B>(A) -> B
+type Bar = <X, Y>(X) -> Y
+local x: Foo = something :: Bar
+
+-- names of parameters are ignored
+type Red = (foo: number?, bar: number?) -> number
+type Blue = (frob: number?, garb: number?) -> number
+local x: Red = something :: Blue
+```
+
+So, there is no reason why named function type returns would be any different. 
```

Comment:
I'd push back against the idea that: if a feature has already been merged in, then parallel features get a free pass.

Reply:
For sure. I don't mean to imply that at all. I'm saying that named returns wouldn't change the precedent that's already set by the existing features, aka. it has aligned/consistent behaviour, not that it should get a free pass.

Reply:
Edited language in doc to be clearer